### DISCUSSION TITLE
Сustom modal window by clicking on a table cell

### DISCUSSION THREAD

**Discussion by Aliakseyeu** [filamentphp/filament#3127] [Help]
Package
filament/tables

Package Version
v2.0

Laravel Version
v9.19

Livewire Version
v2.10

PHP Version
PHP 8.0.2

Question description
Hi all! How can I open a custom modal window by clicking on a table cell? The same as when clicking on an action. I found this in the documentation: 
TextColumn::make('title')
    ->action(function (Post $record): void {
        $this->dispatchBrowserEvent('open-post-edit-modal', [
            'post' => $record->getKey(),
        ]);
    })
But I do not understand how I can customize this piece of code for my needs. Add window, form and handler. Can someone help me?

**Comment by xavvey** [comment]
https://filamentphp.com/docs/2.x/admin/pages/actions#custom-forms
for custom modals

and
https://filamentphp.com/docs/2.x/tables/getting-started#record-urls-clickable-rows
for clickable rows

I think this is what you need?

**Comment by Aliakseyeu** [comment]
Thanks for the reply, but that's not quite right. The first link contains information about creating an action. And it works great. At the end of each row a button is appears with a call to the modal window. But, unfortunately, not for a specific cell. And in the second transition to the URL by clicking on the row. I would like everything to be the same as in the first link, but on a click on the cell, and not on the button at the end of the row

**Reply by xavvey** [reply]
Ah I see what you mean now.

the `$this->dispatchBrowserEvent() `method is from Livewire and not Filament.
https://laravel-livewire.com/docs/2.x/events#browser 

This is the last thing I can think of.. Haven't worked with this myself so don't fully know how to customize it but it might help you a bit

**Reply by Aliakseyeu** [reply]
Ok, thanks for the help. I will dig in the direction of events.
